#! /bin/sh /usr/share/dpatch/dpatch-run
## 25_url-if-no-smartbookmarks.dpatch by Yavor Doganov <yavor@gnu.org>
##
## DP: Don't fail miserable if all smart bookmarks are deleted.
## DP: Workaround for #442448.

@DPATCH@
diff -urNad kazehakase-0.5.6~/src/bookmarks/kz-smart-bookmark.c kazehakase-0.5.6/src/bookmarks/kz-smart-bookmark.c
--- kazehakase-0.5.6~/src/bookmarks/kz-smart-bookmark.c	2009-08-17 17:24:43.000000000 +0300
+++ kazehakase-0.5.6/src/bookmarks/kz-smart-bookmark.c	2009-08-19 23:19:55.000000000 +0300
@@ -252,7 +252,7 @@
 	GMatchInfo *match_info = NULL;
 	GError *error = NULL;
 
-	g_return_val_if_fail(KZ_IS_BOOKMARK(bookmark), NULL);
+	g_return_val_if_fail(KZ_IS_BOOKMARK(bookmark), g_strdup(text));
 
 	if (!text) 
 		return g_strdup("");
